### DISCUSSION TITLE
Why validation doesn't return error directly on change

### DISCUSSION THREAD

**Discussion by Et7f3** [react-hook-form/react-hook-form#11575] [Q&A] [answered]
Here is a repro:
https://stackblitz.com/edit/vitejs-vite-mde7iv?file=src%2FApp.tsx

When I type something like `a`, `b`, `c`, `d`, `↹` in age field. I don't see any error message printed with `console.log` ? I expected that if their is some validation error it will force a refresh (so I could add a red border for instance).
Then when I type `⏎` now for each key I type will trigger the errors until the value become valid.

So `⇤` (`abc`), `⇤` (`ab`), `⇤` (`a`) and `⇤` (``) both trigger refresh. Then If I add valid number no issue but the moment when I add any non digit char the errors re-appear (The expected behaviour).

So what I miss so that error also appear in the first place without requiring user to type `⏎`

**Comment by Moshyfawn** [comment, marked as answer]
If you want your validation to run on input change, you should add the appropriate validation mode value:

```js
useForm({
  mode: "onChange"
})
```

Be sure to check out [the `useForm` docs](https://react-hook-form.com/docs/useform#mode) to learn more.
